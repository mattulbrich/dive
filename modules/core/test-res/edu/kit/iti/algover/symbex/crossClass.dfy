class C {
  var d : D;

  method setD(x : D)
    requires x != null
    ensures d == x
    ensures x.c == this
    modifies this, x
  {
    this.d := x;
    d := x;
    x.c := this;
  }
}

class D {
  var c : C;
}
