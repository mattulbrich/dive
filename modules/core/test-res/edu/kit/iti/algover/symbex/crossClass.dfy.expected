(COMPILATION_UNIT
 (class C
        (FIELD d (TYPE D))
        (method setD (ARGS (VAR x (TYPE D)))
                (requires (!= x null))
                (ensures (== d x))
                (ensures (== (FIELD_ACCESS x c) this))
                (modifies this x)
                (BLOCK
                 (:= (FIELD_ACCESS this d) x)
                 (:= d x)
                 (:= (FIELD_ACCESS x c) this))))

 (class D (FIELD c (TYPE C))))
